include/sensors/core/tasks/pressure_driver.hpp

```diff
@@ -315,24 +315,32 @@ class MMR920 {
         can_client.send_can_message(
             can::ids::NodeId::host,
             can::messages::Acknowledgment{.message_index = count});
-        for (int i = 0; i < count; i++) {
+        std::array<uint32_t, 14> data{};
+        int i = 0;
+        while (i < count) {
             // send over buffer and then clear buffer values
             // NOLINTNEXTLINE(div-by-zero)
             int current_index =
                 (i + start) % static_cast<int>(SENSOR_BUFFER_SIZE);
-
-            can_client.send_can_message(
-                can::ids::NodeId::host,
-                can::messages::ReadFromSensorResponse{
-                    .message_index = message_index,
-                    .sensor = can::ids::SensorType::pressure,
-                    .sensor_id = sensor_id,
-                    .sensor_data = mmr920::reading_to_fixed_point(
-                        (*sensor_buffer).at(current_index))});
-            if (i % 10 == 0) {
-                // slow it down so the can buffer doesn't choke
-                vtask_hardware_delay(20);
+            uint8_t j = 0;
+            while (j < 14 and i < count) {
```

Comment:
We absolutely need some tests for the fiddly index math.

Reply:
also can use `j < data.size()` or even take this to a range-for that stops early a la
```c++
for (auto& copy_to : data) {
    if (i >= count) {
        break;
    }
    copy_to = mmr920::reading_to_fixed_point(sensor_buffer->at(current_index));
    i++;
    current_index = (i + start) % static_cast<int>(SENSOR_BUFFER_SIZE);
}
```